google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/AddMerchantCenterDynamicRemarketingCampaign.java

```diff
@@ -0,0 +1,408 @@
+      return;
+    }
+
+    try {
+      new AddMerchantCenterDynamicRemarketingCampaign()
+          .runExample(
+              googleAdsClient,
+              params.customerId,
+              params.merchantCenterAccountId,
+              params.campaignBudgetId,
+              params.userListId);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /**
+   * Runs the example.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID in which to create criterion.
+   * @throws GoogleAdsException if an API request failed with one or more service errors.
+   */
+  private void runExample(
+      GoogleAdsClient googleAdsClient,
+      long customerId,
+      long merchantCenterId,
+      long budgetId,
+      long userListId)
+      throws IOException {
+    // Creates a shopping campaign associated with a given merchant center account.
+    String campaignResourceName =
+        createCampaign(googleAdsClient, customerId, merchantCenterId, budgetId);
+    // Creates an ad group for the campaign.
+    String adGroupResourceName = createAdGroup(googleAdsClient, customerId, campaignResourceName);
+    // Creates a dynamic display ad in the ad group.
+    createAd(googleAdsClient, customerId, adGroupResourceName);
+    // Targets a specific user list for remarketing.
+    attachUserList(googleAdsClient, customerId, adGroupResourceName, userListId);
+  }
+
+  /**
+   * Creates a campaign linked to a Merchant Center product feed.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID in which to create criterion.
+   * @param merchantCenterId the merchant center account ID.
+   * @param budgetId the campaign budget ID.
+   * @throws GoogleAdsException if an API request failed with one or more service errors.
+   */
+  private String createCampaign(
+      GoogleAdsClient googleAdsClient, long customerId, long merchantCenterId, long budgetId) {
+    String budgetResourceName = ResourceNames.campaignBudget(customerId, budgetId);
+
+    // Creates the campaign.
+    Campaign campaign =
+        Campaign.newBuilder()
+            .setName(StringValue.of("Shopping campaign #" + System.currentTimeMillis()))
+            .setAdvertisingChannelType(AdvertisingChannelType.DISPLAY)
```

Comment:
Add comment (borrowed from the [AdWords API example](https://github.com/googleads/googleads-java-lib/blob/master/examples/adwords_axis/src/main/java/adwords/axis/v201809/advancedoperations/AddShoppingDynamicRemarketingCampaign.java#L210)):

    // Dynamic remarketing campaigns are only available on the Google Display Network.